release 2.9.0 blog post

orbeckst opened the pull request.

orbeckst committed:
draft for release 2.9.0 blog post

IAlibay commented on _posts/2025-03-11-release-2.9.0.md:
```diff
@@ -0,0 +1,109 @@
+  note that in v3.0 theses modules will be removed completely from the
+  MDAnalysis core library and you will need to use the MDAKits
+  directly.
+* `fasteners` dependency has been switch to `filelock`.
```
```suggestion
* `fasteners` dependency has been switched to `filelock`.
```

IAlibay reviewed (approved):
thanks @orbeckst !

p-j-smith reviewed (approved):
Thank you!

orbeckst committed:
Update _posts/2025-03-11-release-2.9.0.md

Co-authored-by: Irfan Alibay <IAlibay@users.noreply.github.com>

orbeckst commented on _posts/2025-03-11-release-2.9.0.md:
```diff
@@ -0,0 +1,109 @@
+
+Supported Python versions: **3.10, 3.11, 3.12, 3.13**.
+
+Please note that Python 3.13 is limited to [PyPi][] for now, the [conda-forge channel][] installs only provide support for Python 3.10 to 3.12.
```
@hmacdope @IAlibay at the moment there are no py3.13 c-f packages https://anaconda.org/conda-forge/mdanalysis/files?page=1 so we have to keep this line (see https://github.com/MDAnalysis/mdanalysis/issues/4805 ).

Is there a chance that this gets fixed soon (within a day) or should we publish the post as is?

IAlibay replied:
I'm travelling sorry, it's a simple migration if others have time though.

orbeckst replied:
@hmacdope do you know what to do?

orbeckst replied:
Thank you both, we now have 3.13 packages.

orbeckst commented on _posts/2025-03-11-release-2.9.0.md:
```diff
@@ -0,0 +1,109 @@
+MDAnalysis thanks [NumFOCUS][] for its continued support as our fiscal sponsor and 
+the [Chan Zuckerberg Initiative][] for supporting MDAnalysis under EOSS4 and EOSS5 awards.
+
+— @IAlibay and @p-j-smith ([release manager][roles]) on behalf of the [MDAnalysis Team][]
```
```suggestion
— @IAlibay and @p-j-smith ([release managers][roles]) on behalf of the [MDAnalysis Team][]
```

orbeckst committed:
Update _posts/2025-03-11-release-2.9.0.md

RMeli commented on _posts/2025-03-11-release-2.9.0.md:
```diff
@@ -0,0 +1,109 @@
+
+This is a minor release of the MDAnalysis library, which means that it
+contains enhancements, bug fixes, deprecations, and other
+backwards-compatible changes.
```
Is this actually true for this one? Should we mention https://github.com/MDAnalysis/mdanalysis/issues/4899?

RMeli replied:
I think it's fine, thanks.

orbeckst committed:
fixes (after rendering locally)

orbeckst committed:
release 2.9.0 also has conda-forge packages for python 3.13

(https://github.com/MDAnalysis/mdanalysis/issues/4805)

(or it will have them soon enough so that I can remove the limitation)

orbeckst commented:
@hmacdope should we emphasize more that people should try and install
```
mamba install distopia
```
to make use of the much faster distance calculations? Or should we include it as a conda package run time dependency for x86 architectures?

orbeckst commented:
Distopia works on my old x86 macOS:
```python
In [1]: import MDAnalysis as mda; import MDAnalysisTests.datafiles as data

In [2]: u = mda.Universe(data.TPR, data.TRR)

In [3]: u.atoms.n_atoms
Out[3]: 47681

In [10]: %timeit d = mda.lib.distances.distance_array(u.atoms[:5000], u.atoms[500:4000],
       ⋮ box=u.dimensions)
629 ms ± 2.78 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)

In [11]: %timeit d = mda.lib.distances.distance_array(u.atoms[:5000], u.atoms[500:4000],
       ⋮ box=u.dimensions, backend="distopia")
324 ms ± 163 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)
```
and gives 2x speed-up for a medium-sized problem.

For all 47k atoms ... 
```python
In [14]: %timeit d = mda.lib.distances.distance_array(u.atoms, u.atoms, box=u.dimensions, backend="distopia")
Segmentation fault: 11
```
oops.... but in all fairness, the serial version ran my laptop out of its 16 GB of memory but completed
```python
In [4]: %timeit d = mda.lib.distances.distance_array(u.atoms, u.atoms, box=u.dimensions,
      ⋮ backend="serial")
1min 36s ± 3.02 s per loop (mean ± std. dev. of 7 runs, 1 loop each)
```

orbeckst commented:
The post could now be published in principle but let’s wait until at least tomorrow in case anyone wants to make any changes.

jennaswa commented on _posts/2025-03-11-release-2.9.0.md:
```diff
@@ -0,0 +1,108 @@
+
+## Notable changes
+
+For a full list of changes, bugfixes and deprecations see the [CHANGELOG][].
```
```suggestion
For a full list of changes, bug fixes and deprecations see the [CHANGELOG][].
```

orbeckst committed:
Update _posts/2025-03-11-release-2.9.0.md

Co-authored-by: Jenna M Swarthout Goddard <38287809+jennaswa@users.noreply.github.com>

orbeckst committed:
highlight distopia

- installation instructions
- link to docs for backends
- mini example
- mention (currently undocumented) MDAnalysis.lib.distances.HAS_DISTOPIA

orbeckst commented on _posts/2025-03-11-release-2.9.0.md:
```diff
@@ -0,0 +1,135 @@
+
+* Support for Gromacs v2024.4 and v2025 TPR files.
+* A new "water" keyword for selecting water residues.
+* Support for **[distopia][] 0.4.0** as an optional distance library
```
@hmacdope can you look over the distopia part, please?

orbeckst replied:
see also PR https://github.com/MDAnalysis/mdanalysis/pull/4970 for doc update

orbeckst commented:
@jennaswa with release post up, are you able to put out announcements (LinkedIn, bsky, discussions)? (Not sure if I have any time soon)
